metals/src/main/scala/scala/meta/metals/ScalametaServices.scala

```diff
@@ -305,11 +313,13 @@ class MetalsServices(
       ()
     }
     .request(td.documentHighlight) { params =>
-      DocumentHighlightProvider.highlight(
-        symbolIndex,
-        Uri(params.textDocument.uri),
-        params.position
-      )
+      if (latestConfig().highlight.enabled) {
```

Comment:
Have you tried to dynamically unregister capabilities? I think it makes sense to keep this guard regardless because I'm not sure if we can rely on it in case a client like vim/sublime to respect dynamic capabilities. Just curious to see if it works.

Reply:
Sorry, didn't get what you're saying here. Do you really mean dynamically unregister capabilities or you mean just enable/disable this setting in the config?

Reply:
I was thinking of using `client/unregisterCapability` https://microsoft.github.io/language-server-protocol/specification#client_unregisterCapability

I think it's best to do it like here in the PR because dynamic capabilities may not be supported by all clients, but I was just wondering if you had tried to used `unregisterCapabilities`

Reply:
Ah, now I understand: when some feature gets enabled/disabled in the config, we should register/unregister the corresponding capability. 

I agree that it's correct, but we don't register capabilities dynamically yet and you're right that it's not widely supported (Atom doesn't support it, for instance), so I'd suggest we address it later, probably after #163.

Reply:
I agree this should be explored later, esp. if it's not even supported by atom 👍